mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -92,10 +92,7 @@ def on_train_start(self, trainer, pl_module):
                 try_mlflow_log(mlflow.log_param, "optimizer_name", type(optimizer).__name__)
 
                 if hasattr(optimizer, "defaults"):
-                    optim_dict = optimizer.defaults
-
-                    for key, value in optim_dict.items():
-                        try_mlflow_log(mlflow.log_param, key, value)
+                    try_mlflow_log(mlflow.log_params, optimizer.defaults)
```

Comment:
Currently, all the default parameters in the optimizer are logged. 

We need to revisit the following scenarios

1. Matching fixed keys as below, restricts the number of parameters. For ex: if user chooses a different optimizer and the optimizer dictionary contains extra keys (apart from `learning_rate`, `betas`, `weight_decay`, `epsilon`) such parameters would be missed to log.

``` 
if "lr" in optmizer.defaults:
        try_mlflow_log(mlflow.log_param, "learning_rate", optimzer.defaults["lr"])
```

Hence the entire default dictionary parameters are stored into mlflow (however , the names are too short `lr` - `learning_rate`)

2. Handling multiple optimizers - As per the current approach only the first optimizer parameters are logged. In case of multiple optimizers, we might need to iterate and find an effective way of logging the parameters (and differentiating the parameters among the optimizers)